### PR TITLE
Replace static keywords with locale lookups in views

### PR DESCRIPTION
**Pull request by njsfield**
Fixes :
  #101 
  #100 

Awaiting review & merge of #92 first (logo change)

## Overview

- Populate en/chn.json files with static keys used in views (chn.json has placeholder characters)
- Refactor several view files and mixins for locale support
- Remove titles from injection into views (from routes)

Every view is assigned a key of `__`, containing the corresponding JSON object data (converted to JS object).
Static fields are available via dot notation, and JSON formatting closely resembles view/partial name, to simplify lookups

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -35,14 +38,56 @@ const routes = {
...
+  // Use express's router as middleware
+  // Base routes are accessible with /{locale}/ prepended
+  const viewRouter = require('express').Router();
+
+  viewRouter.use((req, res, next) => {
+    // Extract locale (e.g.) 'en' or 'chn' from full url
+    const locale = req.originalUrl.split('/')[1];
+
+    // If /en/ or /chn/ not included in url
+    if ([ 'en', 'chn', ].includes(locale) != true) {
```
Comment (emilyb7) [inline routes/index.js]: didn't you put these in the setup file? I guess that's another branch though. Anyway, equality again. 
Or you could just do `(![ 'en', 'chn', ].includes(locale))`, right?

File: templates/partials/header.pug [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+  a(href=`/${locale}` title="Home").no-underline
     #strapline.h-100.self-start.ttu.f7.f5-ns.fw6.light-gray.pa3-ns.pa1.pointer
-      p.ma0 Chinese Film &
-      p.ma0 Medical Humanities
-  #logo.h-100.self-end
-    a.h-100.w-100.fr(href="/" title="YiMovi home")
+      p.ma0 #{__.header.strapline_1}
+      p.ma0 #{__.header.strapline_2}
+  #actions.h-100.self-end.inline-flex.flex-row.items-center
+    a(href=`${new_locale_url}` title="Switch to English").no-underline.white.pointer.ph1 SWITCH TO #{locale_flipped_formatted}
```
Comment (emilyb7) [inline templates/partials/header.pug]: this button needs to be translated too!

File: templates/partials/header.pug [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
+    a.h-100.w-100.fr(href=`/${locale}` title="YiMovi home")
       .logo.h-100
         include ../partials/logo.svg
```
Comment (emilyb7) [inline templates/partials/header.pug]: the logo is no longer aligned against the edge of the page

File: templates/partials/header.pug (hunk 2 above)
Comment (emilyb7) [reply templates/partials/header.pug]: also let's make this look better, yeah? I have some ideas of how to style this if you don't...

**Review by emilyb7 — CHANGES_REQUESTED** [review]

File: templates/partials/header.pug (hunk 2 above)
Comment (njsfield) [reply templates/partials/header.pug]: What are your suggestions?

Comment (emilyb7) [reply templates/partials/header.pug]: here are a couple of examples off the top of my head:

- keystone's own docs: http://keystonejs.com/docs/
- https://www.theguardian.com/uk

I think we could have it even smaller though. Just a simple switch like `CHN / EN` and whatever locale you're currently viewing, that word is highlighted.

**Comment by emilyb7** [comment]
if i'm viewing `en` locale and click on the logo in the header, I now get redirected to the homepage for `chn`

**Review by emilyb7 — CHANGES_REQUESTED** [review]
see latest comment

**Comment by njsfield** [comment]
@emilyb7 I cant replicate this. Is this occurring for you in this branch?

**Comment by njsfield** [comment]
Update: imported changes in PR #92

**Comment by emilyb7** [comment]
cool, it all seems to be working now. But let's get the smaller PRs merged in before this massive one!

**Pull request closed without merging**